transaction-signer/src/lib.rs

```diff
@@ -0,0 +1,316 @@
+use bip39::{Language, Mnemonic, MnemonicType};
+use mc_account_keys::AccountKey;
+use mc_common::HashMap;
+use mc_core::slip10::Slip10KeyGenerator;
+use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic};
+use mc_full_service::{
+    db::{account::AccountID, txo::TxoID},
+    json_rpc::{
+        json_rpc_request::JsonRPCRequest,
+        v2::{
+            api::request::JsonCommandRequest,
+            models::{
+                account_key::AccountKey as AccountKeyJSON,
+                account_secrets::AccountSecrets,
+                tx_proposal::{
+                    TxProposal as TxProposalJSON, UnsignedTxProposal as UnsignedTxProposalJSON,
+                },
+            },
+        },
+    },
+    service::models::tx_proposal::UnsignedTxProposal,
+    util::encoding_helpers::{ristretto_public_to_hex, ristretto_to_hex},
+};
+use mc_transaction_core::{
+    get_tx_out_shared_secret,
+    onetime_keys::{recover_onetime_private_key, recover_public_subaddress_spend_key},
+    ring_signature::KeyImage,
+    tx::TxOut,
+};
+use std::{convert::TryFrom, fs};
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+
+#[pymodule]
```

Comment:
I don't think we should be testing Rust code using something that exposes it to Python.
If the goal is to call actual rust functions for tests, the test should be written in Rust.
If the goal is to test the Rust code as a block box, the Python code should run the binary like regular users are expected to run it.
Nothing is currently expected to use these bindings, so exercising them in a test is in my opinion a questionable approach and I'm not sure how much work it saves us, compared to just running the binary.

Reply:
yeah i just took the simpler approach and am about to PR that